### ISSUE TITLE
Can't link to files inside Media folder

### ISSUE THREAD

**Issue by GRMacGeek** [yuzoolcode/pulsecms-tracking#31]
Okay if I upload an image or pdf into the Media folder, and then try to set a link to it from within a block, the "pulse" part of the URL gets rewritten as "???"

For example

- there is a pdf here [https://www.grandvilleeducationfoundation.org/pulse/content/media/pdf/Exec-Director-Job-Description-2-2018.pdf](https://www.grandvilleeducationfoundation.org/pulse/content/media/pdf/Exec-Director-Job-Description-2-2018.pdf)
 but when I link to it from within redactor, the result is this [https://www.grandvilleeducationfoundation.org/???/content/media/pdf/Exec-Director-Job-Description-2-2018.pdf](https://www.grandvilleeducationfoundation.org/???/content/media/pdf/Exec-Director-Job-Description-2-2018.pdf)

You can see this in action on this [hidden page](https://www.grandvilleeducationfoundation.org/test/) and half way down the page I tried to add a link for "Executive Director Job Description here" 
My work around on the "live" page was to put the pdf in a different folder outside of Pulse and link to that, but my clients need to be able to add media and link to it...

**Comment by jrondorf** [comment]
I have seen this before. This ??? is only generated inside the RW page, or also in a Pulse page?
How did you inserted the link in Redactor?

**Comment by GRMacGeek** [comment]
Link entered in Redactor, and looks fine even when you view the html See attached images
![pulse_cms](https://user-images.githubusercontent.com/6706155/36617362-0dec6b06-18b5-11e8-9795-98ec3e42198f.jpg)
![pulse_cms2](https://user-images.githubusercontent.com/6706155/36617365-10a9933c-18b5-11e8-8e03-c5d0de5b4483.jpg)

But when you view it live on the website the "pulse" part of the link gets changed to "???" for some reason...
 I tried it with a few different media files, and always the same result.

**Comment by jrondorf** [comment]
Again, is it in your case only rendered inside RW wrong, or also on a plain pulse page?

**Comment by GRMacGeek** [comment]
Okay not sure what you are trying to ask here...

Yes this is a Pulse block in Rapidweaver. But it can't obviously "render" in Rapidweaver, the issue happens when you view the "live" link on the website. Even though the link is correct in the block, and stays correct even after I save it, it changes on the live website... test page ["rendered" here](https://www.grandvilleeducationfoundation.org/test/)

**Comment by jrondorf** [comment]
I am trying to find out if it is a general pulse problem, or if the problem only is related to a block displayed inside RW. 

What happens if you put that block with the link in a Pulse page and view it in http://domain/pulse/page ?

**Comment by GRMacGeek** [comment]
How do I add a "Pulse Page" to a Rapidweaver project?

**Comment by jrondorf** [comment]
OK, you don't get it.

FYI: You are able to use Pulse without RapidWeaver...

And yes, you are also able to add whole Pulse Pages to a RapidWeaver project. RTFM...

**Comment by jrondorf** [comment]
@yuzoolcode This is a bug in block.txt generation. After you enter the link in the admin dashboard, it gets corrupted.

When adding a link like this:

![2018-02-24_14-32-10](https://user-images.githubusercontent.com/2099795/36630980-3ea42da4-1970-11e8-8e14-4ba68b960b06.png)

And the block looks like this:

![2018-02-24_14-32-25](https://user-images.githubusercontent.com/2099795/36630983-4f6a7f30-1970-11e8-8a72-a7e79cab8d8c.png)

The block save process generated a wrong link inside the text file. Why is `{{show_var:path}}` automatically inserted here?

`<p>This is the link:</p><p><a href="http:/{{show_var:path}}5.instacks.com{{show_var:path}}/content/media/documents/document.pdf">Document</a><br></p>`

**Comment by GRMacGeek** [comment]
Wow - no kidding, you can use Pulse without Rapidweaver... Do you realize just how rude that sounds? And as for reading the f'ing manual (yet another example of you being rude...) your instructions are very vague and hard to follow in English... as evidenced by other Rapidweaver users having a hard time with them...

**Comment by jrondorf** [comment]
https://instacks.com/pulsecmsstack/tutorial/#page

**Comment by jrondorf** [comment]
No, my instructions aren't hard to read.

Actually, I don't have that many users who aren't able to follow my instructions, just one or two...

**Comment by jrondorf** [comment]
In addition, I am not rude. RTFM is a common term in software development:

https://en.m.wikipedia.org/wiki/RTFM

**Comment by jrondorf** [comment]
Fixed for 5.1. Thanks for retest!
